server/reframe/reframe.go

```diff
@@ -0,0 +1,70 @@
+package reframe
+
+import (
+	"bytes"
+	"net/http"
+
+	"github.com/filecoin-project/go-indexer-core"
+	"github.com/filecoin-project/storetheindex/internal/registry"
+	"github.com/filecoin-project/storetheindex/server/finder/handler"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-delegated-routing/client"
+	"github.com/ipfs/go-delegated-routing/server"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/routing"
+	"github.com/multiformats/go-multicodec"
+	"github.com/multiformats/go-multihash"
+	"github.com/multiformats/go-varint"
+)
+
+func NewReframeHTTPHandler(indexer indexer.Interface, registry *registry.Registry) http.HandlerFunc {
+	return server.DelegatedRoutingAsyncHandler(NewReframeService(handler.NewFinderHandler(indexer, registry)))
+}
+
+func NewReframeService(fh *handler.FinderHandler) *ReframeService {
+	return &ReframeService{finderHandler: fh}
+}
+
+type ReframeService struct {
+	finderHandler *handler.FinderHandler
+}
+
+func (x *ReframeService) FindProviders(key cid.Cid) (<-chan client.FindProvidersAsyncResult, error) {
+	mh := key.Hash()
+	fr, err := x.finderHandler.Find([]multihash.Multihash{mh})
+	if err != nil {
+		return nil, err
+	}
+	ch := make(chan client.FindProvidersAsyncResult)
+	peerAddrs := []peer.AddrInfo{}
+	for _, mhr := range fr.MultihashResults {
+		if !bytes.Equal(mhr.Multihash, mh) {
+			continue
```

Comment:
Should an error be logged here, since there should never be a result for a multihash that was not asked for?

Reply:
The type ReframeService is just a pass-through adapter on top of FinderHandler. Therefore it should not be used-for/relied-on for discovering bugs in the underlying FinderHandler. I am happy to add a log, but if you want to catch bugs in the FinderHandler, the logging should be there. Let me know what you think.